Integrate QIT security and malware into PR checks

RadoslavGeorgiev opened the pull request:
Fixes WOOPMNT-4936

#### Changes proposed in this Pull Request

- [x] Add a new `npm run qit:malware` command, together with the needed script.
- [x] Add CI checks

#### Testing instructions

1. Following the instructions in `tests/qit/README.md`.
2. Run `npm run qit:malware` (this is new, security was already there).
3. Verify that it works, as well as the workflows on this PR.

RadoslavGeorgiev committed:
Adding a malware script

RadoslavGeorgiev committed:
Add an npm command

botwoo commented:
### Test the build 

#### Option 1. Jetpack Beta

- Install and activate [Jetpack Beta](https://jetpack.com/download-jetpack-beta/).
- Use this build by searching for PR number `10729` or branch name `WOOPMNT-4936` in your-test.site/wp-admin/admin.php?page=jetpack-beta&plugin=woocommerce-payments          

#### Option 2. Jurassic Ninja - available for logged-in A12s

:rocket: [Launch a JN site with this branch](https://jurassic.ninja/create/?jetpack-beta&shortlived&nojetpack&woocommerce&woocommerce-payments-dev-tools&branches.woocommerce-payments=WOOPMNT-4936) :rocket:

:information_source: Install this [Tampermonkey script](https://github.com/Automattic/woocommerce-payments/tree/develop/bin/wcpay-live-branches) to get more options.

---

Build info: 

- Latest commit: 0241a0fa082141b70c53f81ea77f900e3b52c891
- Build time: 2025-04-29 21:30:05 UTC

_Note: the build is updated when a new commit is pushed to this PR._

RadoslavGeorgiev committed:
Initial QIT action

RadoslavGeorgiev committed:
Simplified action

There are some credentials in GitHub already, let's see if they work.

Example was used from https://qit.woo.com/docs/using-qit/github-workflows/

RadoslavGeorgiev committed:
Remove the authentication step, set environment variables instead.

The local scripts already use env variables to authenticate. Rather than re-doing that, just provide env variables.

RadoslavGeorgiev committed:
Do not require the composer package globally

RadoslavGeorgiev committed:
Switch back to the global composer version.

RadoslavGeorgiev committed:
Changelog entry

frosso reviewed (approved):
Excellent job!
I see the successful run: https://github.com/Automattic/woocommerce-payments/actions/runs/14741650518?pr=10729

I also ran it locally to test it.
<img width="362" alt="Screenshot 2025-04-30 at 9 11 43 AM" src="https://github.com/user-attachments/assets/1a549852-2284-4451-ab73-88e803844e39" />

Works perfectly :shipit:
